velox/row/UnsafeRowDeserializers.cpp

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/row/UnsafeRowDeserializers.h"
+
+namespace facebook::velox::row {
+namespace {
+
+inline int64_t getFieldOffset(int64_t nullBitsetWidthInBytes, int32_t index) {
+  return nullBitsetWidthInBytes + UnsafeRow::kFieldWidthBytes * index;
+}
+
+inline bool isNullAt(const uint8_t* memoryAddress, int32_t index) {
+  return bits::isBitSet(memoryAddress, index);
+}
+
+int32_t getTotalStringSize(
+    int32_t columnIdx,
+    int32_t numRows,
```

Comment:
It seems numRows is 'vector_size_t' type in the 'deserialize' method. Perhaps also fix this in the createFlatVectorFast functions.